packages/frontend/src/hooks/useFathomTrackPageWatch.ts

```diff
@@ -0,0 +1,13 @@
+import { useFathom } from "@carrot-kpi/react";
+import { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+
+export const useFathomTrackPageWatch = () => {
+    const location = useLocation();
+    const fathom = useFathom();
+
+    useEffect(() => {
+        if (!fathom) return;
+        fathom.trackPageview({ url: location.pathname });
```

Comment:
Does this react to a network switch? Do we want to track network switches too?

Reply:
No, currently this does not react to the network switch since it's just the query param that changes and not the actual path; anyway I don't know if it makes sense to track network switches (for now, maybe in the future).